librz/core/cbin.c

```diff
@@ -2705,6 +2709,40 @@ static bool strings_print(RzCore *core, RzCmdStateOutput *state, const RzPVector
 		char quiet_val[20];
 		ut64 paddr, vaddr;
 		paddr = string->paddr;
+
+		char *xref_address_str = NULL;
+		char *flag_name_str = NULL;
+		if (print_xref) {
+			// search xrefs to string
+			RzList *xrefs = rz_analysis_xrefs_get_to(core->analysis, paddr);
+			RzAnalysisXRef *xref;
+			RzListIter *xrefIter;
+			RzStrBuf *xref_addresses = rz_strbuf_new("");
+			rz_list_foreach (xrefs, xrefIter, xref) {
+				char addr_str[32];
+				rz_strf(addr_str, "0x%08" PFMT64x, xref->from);
+				if (rz_strbuf_length(xref_addresses) > 0) {
+					rz_strbuf_append(xref_addresses, ",");
+				}
+				rz_strbuf_append(xref_addresses, addr_str);
+			}
+			rz_list_free(xrefs);
+			xref_address_str = rz_strbuf_drain(xref_addresses);
+
+			// search possible flag
+			const RzList *flags = rz_flag_get_list(core->flags, paddr);
+			RzFlagItem *flag;
+			RzListIter *flagIter;
+			RzStrBuf *flag_names = rz_strbuf_new("");
+			rz_list_foreach (flags, flagIter, flag) {
+				if (rz_strbuf_length(flag_names) > 0) {
+					rz_strbuf_append(flag_names, ",");
+				}
+				rz_strbuf_append(flag_names, flag->name);
+			}
+			flag_name_str = rz_strbuf_drain(flag_names);
```

Comment:
Please move this whole case into a `static` function. The `strings_print` gets too long otherwise.